### ISSUE TITLE
Task exceptions are not forwarded to caller

### ISSUE THREAD

**Issue by fmoessbauer** [bshoshany/thread-pool#14]
In case an exception is thrown inside a task, this exception leads to the direct termination of the application.
To fix that, the scheduler has to catch the exception in the sub-task and set the exception-context of the promise object itself.
For details, see cppreference: https://en.cppreference.com/w/cpp/thread/promise/set_exception

Example:

```cpp
auto fut = pool.submit([](){throw std::runtime_error("error");});
try {
  fut.get();
} catch(std::runtime_error & e){
  std::cout << "catched error" << std::endl;
}
```

This application terminates with:

```
terminate called after throwing an instance of 'std::runtime_error'
```

**Comment by bshoshany** [comment]
Thanks for the suggestion! I will fix this in the next release.

**Comment by bshoshany** [comment]
Update: This has now been fixed in v1.8!
